### ISSUE TITLE
[java] StringInstatiation: False negative with String-array access

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2141]
**Affects PMD Version:** 
Running 6.19.0, 6.20.0.

**Rule:**
[StringInstantiation](https://pmd.github.io/latest/pmd_rules_java_performance.html#stringinstantiation)

**Description:**
The rule does not catch String instantiation of array subscripted variable.

**Code Sample demonstrating the issue:**

```java
package pmdtests;

public final class TestStringInstantiation {

  private TestStringInstantiation() {
    throw new IllegalStateException("Private constructor");
  }

  public static void main(String... args) {
    String[] vals = splitOn(args,2,"=");
    for (String val: vals)
      System.out.println(val);
 }
  
private static String[] splitOn(String[] values,int index, String separator) {
    return new String(values[index]).split(separator);
  }

}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
